lib/records.ncl

```diff
@@ -296,4 +296,57 @@ let error_lib = import "./error.ncl" in
                 }
             )
             'Ok,
+
+  contract = {
+    maxProperties
+      | Number -> Dyn
```

Comment:
Same question (I don't have a definite answer at this point, just a question): do we want to pay a runtime cost for this? Given that this is a function that is supposed to be used by generated code, and not actually called by user code?